packages/volto/src/components/theme/View/AlbumView.jsx

```diff
@@ -18,178 +13,138 @@ import openSVG from '@plone/volto/icons/open.svg';
 import aheadSVG from '@plone/volto/icons/ahead.svg';
 import backSVG from '@plone/volto/icons/back.svg';
 
-/**
- * Album view component class.
- * @function AlbumView
- * @param {Object} content Content object.
- * @returns {string} Markup of the component.
- */
-class AlbumView extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      openIndex: undefined,
-    };
-
-    this.closeModal = this.closeModal.bind(this);
-    this.nextImage = this.nextImage.bind(this);
-    this.prevImage = this.prevImage.bind(this);
-  }
-
-  closeModal() {
-    this.setState({
-      openIndex: -1,
-    });
-  }
-
-  nextImage() {
-    const openIndex =
-      (this.state.openIndex + 1) % this.props.content.items.length;
-    this.setState({
-      openIndex,
-    });
-  }
-
-  prevImage() {
-    const openIndex =
-      (this.state.openIndex - 1) % this.props.content.items.length;
-    this.setState({
-      openIndex,
-    });
-  }
+const AlbumView = ({ content }) => {
+  const [openIndex, setopenIndex] = useState(undefined);
 
-  render() {
-    const { content } = this.props;
-    const Container =
-      config.getComponent({ name: 'Container' }).component || SemanticContainer;
-    const PreviewImage = config.getComponent({
-      name: 'PreviewImage',
-    }).component;
+  const closeModal = () => {
+    setopenIndex(-1);
+  };
 
-    return (
-      <Container className="view-wrapper">
-        <article id="content">
-          <header>
-            <h1 className="documentFirstHeading">{content.title}</h1>
-            {content.description && (
-              <p className="documentDescription">{content.description}</p>
-            )}
-          </header>
-          <section id="content-core">
-            <Grid doubling stackable columns={4}>
-              {content.items &&
-                content.items.map((item, index) => (
-                  <React.Fragment key={item.url}>
-                    {item.image_field && (
-                      <Modal
-                        className="gallery"
-                        onClose={this.closeModal}
-                        open={this.state.openIndex === index}
-                        trigger={
-                          <Grid.Column>
-                            <Segment className="imageborder">
+  const nextImage = () => {
+    let openIndexes = (openIndex + 1) % content.items.length;
```

Comment:
This expression gives us a single new value, so it is confusing to give it a plural name.